data/scripts.js

```diff
@@ -140,7 +140,10 @@ exports.BattleScripts = {
 				this.add('-notarget');
 				return true;
 			}
-			if (targets.length > 1) move.spreadHit = true;
+			if (targets.length > 1) {
+				this.attrLastMove('[spread]');
+				move.spreadHit = true;
+			}
```

Comment:
The only place I can find where `p1a`/`p1b`/others are made is in `BattlePokemon#toString` and it doesn't seem to be stored anywhere else. Does that mean `targets[i].toString().substr(0, 3)` (or something along those lines) is the only way to get the location ID?